src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -198,32 +199,96 @@ public Consumer create(Consumer entity, boolean flush) {
 
     @Override
     @Transactional
-    public void delete(Consumer entity) {
-        log.debug("Deleting consumer: {}", entity);
+    public void delete(Consumer consumer) {
+        log.debug("Deleting consumer: {}", consumer);
 
         // Fetch the principal that's triggering this
         Principal principal = this.principalProvider.get();
 
-        Owner owner = entity.getOwner();
-
         // Check if we've already got a record for this consumer (???), creating one if necessary
+        DeletedConsumer deletedConsumer = getDeletedConsumer(consumer);
+        populateEntity(deletedConsumer, consumer.getUuid(), principal, consumer.getOwner());
+
+        super.delete(consumer);
+        this.deletedConsumerCurator.saveOrUpdate(deletedConsumer);
+    }
+
+    private DeletedConsumer getDeletedConsumer(Consumer entity) {
         DeletedConsumer deletedConsumer = this.deletedConsumerCurator.findByConsumerUuid(entity.getUuid());
         if (deletedConsumer == null) {
             deletedConsumer = new DeletedConsumer();
         }
+        return deletedConsumer;
+    }
 
-        // Set/update the properties on our deleted consumer record
-        deletedConsumer.setConsumerUuid(entity.getUuid())
-            .setOwnerId(entity.getOwnerId())
+    private DeletedConsumer populateEntity(DeletedConsumer deletedConsumer, String uuid,
+        Principal principal, Owner owner) {
+        return deletedConsumer.setConsumerUuid(uuid)
+            .setOwnerId(owner.getOwnerId())
             .setOwnerKey(owner.getKey())
             .setOwnerDisplayName(owner.getDisplayName())
             .setPrincipalName(principal != null ? principal.getName() : null);
+    }
 
-        // Actually delete the consumer
-        super.delete(entity);
+    @Override
+    @Transactional
+    public void bulkDelete(Collection<Consumer> consumers) {
```

Comment:
There's a lot of abstraction going on here that doesn't really help us and makes this logic difficult to read through. In general, if this is the direction we're going to go on a given implementation, I'd like to see the methods using each other closer together and in a consistent declaration order, and have the abstractions we do need reasonably minimized.

In this particular case, we can save a lot of unnecessary stack churn and method hopping by compressing the one-off one-liner stuff back into where they're called and combining the other abstractions into a pair of methods (with one nested in the other). That would leave us with something like this:

```
@Override
public void delete(Consumer consumer) {
    log.debug("Deleting consumer: {}", consumer);
    this.bulkDelete(Collections.singleton(consumer));
}

@Override
@Transactional
public void bulkDelete(Collection<Consumer> consumers) {
    if (consumers == null || consumers.isEmpty()) {
        return;
    }

    log.debug("Deleting {} consumer(s)", consumers.size());

    String<String, Consumer> consumerMap = consumers.stream()
        .collect(Collectors.toMap(Consumer::getUuid, Function.identity()));

    int deleted = this.deleteByUuids(consumerMap.keySet());
    log.debug("Deleted {} consumer(s)", deleted);

    Collection<DeletedConsumer> dcrecords = this.buildDeletedConsumerRecords(consumerMap);
    this.deletedConsumerCurator.saveOrUpdateAll(dcrecords.values(), false, false);
}

private int deleteByUuids(Collection<String> uuids) {
    if (uuids == null || uuids.isEmpty()) {
        return 0;
    }

    String jpql = "DELETE Consumer c WHERE c.uuid IN (:uuids)";
    Query query = this.getEntityManager().createQuery(jpql);

    int deleted = 0;
    for (Collection<String> certIdBlock : this.partition(uuids)) {
        deleted += query.setParameter("uuids", certIdBlock)
            .executeUpdate();
    }

    return deleted;
}

private List<DeletedConsumer> buildDeletedConsumerRecords(Map<String, Consumer> consumerMap) {
    Principal principal = this.principalProvider.get();
    String principalName = principal != null ? principal.getName() : null;

    Map<String, DeletedConsumer> dcs = this.deletedConsumerCurator.findByConsumerUuids(consumerMap.keySet())
        .stream()
        .collect(Collectors.toMap(DeletedConsumer::getConsumerUuid, Function.identity()));

    Function<Consumer, DeletedConsumer> mapper = (consumer) -> {
        DeletedConsumer dc = dcs.get(consumer.getUuid());
        if (dc == null) {
            dc = new DeletedConsumer();
        }

        Owner owner = consumer.getOwner();

        return dc.setConsumerUuid(consumer.getUuid())
            .setOwnerId(owner.getOwnerId())
            .setOwnerKey(owner.getKey())
            .setOwnerDisplayName(owner.getDisplayName())
            .setPrincipalName(principalName);
    };

    return consumerMap.values()
        .stream()
        .map(mapper)
        .collect(Collectors.toList());
}
```